src/sql/src/query_model/rewrite_engine.rs

```diff
@@ -965,3 +966,33 @@ impl Rule for WindowingToSelect {
         b.box_type = BoxType::Select(Select::new());
     }
 }
+
+/// A Grouping box without aggregations is a Select box with
+/// [DistinctOperation::Enforce]
+#[derive(Default)]
+struct GroupingToDistinct;
+
+impl Rule for GroupingToDistinct {
+    fn name(&self) -> &'static str {
+        "GroupingToDistinct"
+    }
+
+    fn rule_type(&self) -> RuleType {
+        RuleType::PostOrder
```

Comment:
Is PostOrder right here? I feel this transform doesn't care if it is applied in PreOrder or PostOrder.

Reply:
Yeah, perhaps a forth group of transforms could be added for those transforms that could be applied in any pass of the traversal.